ircd/s_serv.c

```diff
@@ -510,26 +512,37 @@ burst_modes_TS6(struct Client *client_p, struct Channel *chptr,
 {
 	rb_dlink_node *ptr;
 	struct Ban *banptr;
+	char bmask[BANLEN];
 
-	send_multiline_init(client_p, " ", ":%s BMASK %ld %s %c :",
-			me.id,
-			(long)chptr->channelts,
-			chptr->chname,
-			flag);
+	sprintf(buf, ":%s BMASK %ld %s %c",
+		me.id,
+		(long)chptr->channelts,
+		chptr->chname,
+		flag);
+
+	send_multiline_init(client_p, " ", "%s :", buf);
 
 	RB_DLINK_FOREACH_PREV(ptr, list->tail)
 	{
 		banptr = ptr->data;
 
 		if (banptr->forward)
-			send_multiline_item(client_p, "%s$%s",
-					banptr->banstr,
-					banptr->forward);
+			sprintf(bmask, "%s$%s", banptr->banstr, banptr->forward);
+		else
+			strcpy(bmask, banptr->banstr);
+
+		if (IsCapable(client_p, CAP_EBMASK))
+			sendto_one(client_p, "%s %s %ld %s",
+				buf,
+				bmask,
+				(long)banptr->when,
+				banptr->who);
```

Comment:
Even though this will work, I think it is clearer to have a separate command (named EBMASK) with the new format. This avoids the need to check whether the other side thinks we have the EBMASK capab.